### PR TITLE
Add image2image example script.

### PR DESCRIPTION
**Pull request by patil-suraj**
This PR adds an example script for image2image using stable diffusion. 

This PR also re-organizes the `examples` folder, it now contains two sub-directories. One for `inference` and  one for `training`.

@anton-l @patrickvonplaten Happy to hear any suggestions about this :)

### CODE REVIEW

File: examples/inference/readme.md [hunk 1]
```diff
@@ -0,0 +1,50 @@
+# Inference Examples
+
+## Installing the dependencies
+
+Before running the scipts, make sure to install the library's training dependencies:
```
Comment (pcuenca) [inline examples/inference/readme.md:5]:
```suggestion
Before running the scipts, make sure to install the library's dependencies:
```

File: examples/inference/readme.md [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+
+```bash
+pip install diffusers transformers ftfy
+```
+
+## Image-to-Image text-guided generation with Stable Diffusion
+
+The `image_to_image.py` implements `StableDiffusionImg2ImgPipeline`, it let's you pass a text prompt and an initial image to condition on to generate new images. This examples also showcases how you can write custom diffusion pipelines using `diffusers`.
```
Comment (pcuenca) [inline examples/inference/readme.md:13]:
```suggestion
The `image_to_image.py` script implements `StableDiffusionImg2ImgPipeline`. It lets you pass a text prompt and an initial image to condition the generation of new images. This example also showcases how you can write custom diffusion pipelines using `diffusers`!
```

**Review by pcuenca — APPROVED** [review]
I love this pipeline, just left a couple of nits. Also, should we rename `inference/readme.md` to `inference/README.md` for consistency with the training script?

**Review by anton-l — APPROVED** [review]
Awesome pipeline, thank you @patil-suraj! Would be great to link it (or the pipeline examples in general) from the main README too 🚀

File: README.md [hunk 3]
```diff
@@ -23,7 +23,8 @@ More precisely, 🤗 Diffusers offers:
 - State-of-the-art diffusion pipelines that can be run in inference with just a couple of lines of code (see [src/diffusers/pipelines](https://github.com/huggingface/diffusers/tree/main/src/diffusers/pipelines)).
 - Various noise schedulers that can be used interchangeably for the prefered speed vs. quality trade-off in inference (see [src/diffusers/schedulers](https://github.com/huggingface/diffusers/tree/main/src/diffusers/schedulers)).
 - Multiple types of models, such as UNet, can be used as building blocks in an end-to-end diffusion system (see [src/diffusers/models](https://github.com/huggingface/diffusers/tree/main/src/diffusers/models)).
-- Training examples to show how to train the most popular diffusion models (see [examples](https://github.com/huggingface/diffusers/tree/main/examples)).
+- Training examples to show how to train the most popular diffusion models (see [examples/training](https://github.com/huggingface/diffusers/tree/main/examples/training)).
+- Inference examples to show how to create custom pipelines for advanced tasks such as image2image, in-painting (see [examples/inference](https://github.com/huggingface/diffusers/tree/main/examples/inference))
```
Comment (patil-suraj) [inline README.md:27]: @anton-l Linked the examples here.

Comment (anton-l) [reply README.md:27]: Perfect!

**Pull request merged**